File: src/main/java/com/refinedmods/refinedstorage/apiimpl/network/NetworkNodeRegistry.java
package com.refinedmods.refinedstorage.apiimpl.network;

import com.refinedmods.refinedstorage.api.network.node.INetworkNodeFactory;
import com.refinedmods.refinedstorage.api.network.node.INetworkNodeRegistry;
import net.minecraft.resources.ResourceLocation;

import javax.annotation.Nullable;
import java.util.HashMap;
import java.util.Map;

public class NetworkNodeRegistry implements INetworkNodeRegistry {
    private final Map<ResourceLocation, INetworkNodeFactory> factories = new HashMap<>();

    @Override
    public void add(ResourceLocation id, INetworkNodeFactory factory) {
        factories.put(id, factory);
    }

    @Override
    @Nullable
    public INetworkNodeFactory get(ResourceLocation id) {
        return factories.get(id);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/apiimpl/network/NetworkNodeRegistry.kt
package com.refinedmods.refinedstorage.apiimpl.network

import com.refinedmods.refinedstorage.api.network.node.INetworkNodeFactory
import com.refinedmods.refinedstorage.api.network.node.INetworkNodeRegistry
import net.minecraft.util.Identifier

class NetworkNodeRegistry : INetworkNodeRegistry {
    private val factories: MutableMap<Identifier, INetworkNodeFactory> = HashMap()
    override fun add(id: Identifier, factory: INetworkNodeFactory) {
        factories[id] = factory
    }

    override operator fun get(id: Identifier): INetworkNodeFactory? {
        return factories[id]
    }
}